cmd/kepify/main.go

```diff
@@ -0,0 +1,184 @@
+		os.Exit(1)
+	}
+	if len(files) == 0 {
+		fmt.Fprintf(os.Stderr, "did not find any KEPs\n")
+		os.Exit(1)
+	}
+
+	// Parse the files
+	proposals, err := parseFiles(files)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "error parsing files: %q\n", err)
+		os.Exit(1)
+	}
+
+	// Generate the json output
+	err = printJSONOutput(*filePath, proposals)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "could not open file: %v\n", err)
+		os.Exit(1)
+	}
+}
+
+func findMarkdownFiles(dirPath *string) ([]string, error) {
+	files := []string{}
+	err := filepath.Walk(
+		*dirPath,
+		func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if info.IsDir() {
+				return nil
+			}
+			if ignore(info.Name()) {
+				return nil
+			}
+			files = append(files, path)
+			return nil
+		},
+	)
+	return files, err
+}
+
+func parseFiles(files []string) (keps.Proposals, error) {
+	var proposals keps.Proposals
+	for _, filename := range files {
+		parser := &keps.Parser{}
+		file, err := os.Open(filename)
+		if err != nil {
+			return nil, fmt.Errorf("could not open file: %v\n", err)
+		}
+		defer file.Close()
+		kep := parser.Parse(file)
+		// if error is nil we can move on
+		if kep.Error != nil {
+			return nil, fmt.Errorf("%v has an error: %q\n", filename, kep.Error.Error())
+		}
+		fmt.Printf(">>>> parsed file successfully: %s\n", filename)
+		proposals.AddProposal(kep)
+	}
+	return proposals, nil
+}
+
+func printJSONOutput(filePath string, proposals keps.Proposals) error {
+	fmt.Printf("Output file: %s\n", filePath)
+	file, err := os.Create(filePath)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	total := len(proposals)
+	fmt.Printf("Total KEPs: %d\n", total)
+
+	fmt.Fprintln(file, "{")
+	for i, kep := range proposals {
+		fmt.Fprintf(file, "\t\"%s\": {\n", hash(kep.OwningSIG+":"+kep.Title))
+		fmt.Fprintf(file, "\t\t\"%s\": \"%s\",\n", "title", kep.Title)
+		fmt.Fprintf(file, "\t\t\"%s\": \"%s\",\n", "owning-sig", kep.OwningSIG)
+		fmt.Fprintf(file, "\t\t\"%s\": \"%s\",\n", "participating-sigs", strings.Join(kep.ParticipatingSIGs, ","))
+		fmt.Fprintf(file, "\t\t\"%s\": \"%s\",\n", "reviewers", strings.Join(kep.Reviewers, ","))
+		fmt.Fprintf(file, "\t\t\"%s\": \"%s\",\n", "authors", strings.Join(kep.Authors, ","))
+		fmt.Fprintf(file, "\t\t\"%s\": \"%s\",\n", "editor", kep.Editor)
+		fmt.Fprintf(file, "\t\t\"%s\": \"%s\",\n", "creation-date", kep.CreationDate)
+		fmt.Fprintf(file, "\t\t\"%s\": \"%s\",\n", "last-updated", kep.LastUpdated)
+		fmt.Fprintf(file, "\t\t\"%s\": \"%s\",\n", "status", kep.Status)
+		fmt.Fprintf(file, "\t\t\"%s\": \"%s\",\n", "see-also", strings.Join(kep.SeeAlso, ","))
+		fmt.Fprintf(file, "\t\t\"%s\": \"%s\",\n", "replaces", strings.Join(kep.Replaces, ","))
+		fmt.Fprintf(file, "\t\t\"%s\": \"%s\",\n", "superseded-by", strings.Join(kep.SupersededBy, ","))
+		contents, _ := json.Marshal(kep.Contents)
+		fmt.Fprintf(file, "\t\t\"%s\": %s\n", "markdown", contents)
```

Comment:
I think we could somehow split the table of contents from the markdown. This way we easily could preview the KEP only by its TOC.

Reply:
let's do this in a follow up please

Reply:
Okay, good :+1: